### ISSUE TITLE
Assembler(all) waiting state dont count towards X

### ISSUE THREAD

**Issue by zymex22** [zymex22/Project-RimFactory-Revived#118]
**Describe the bug**
When using assemblers eq. Cooker and setting to do X bills (10)
when 10 are made but not placed it will keep making more as it don't count "waiting to be placed" products as "in stock"

Will mostly be present if output is blocked by a wall or other products not being transported away.

**To Reproduce**
Reproducibility: Always
See above

**Expected behavior**
Can be solved by:
- Making waiting to be placed count in stock. (Maybe if we made all the workshops Buildings_Storage?)
- Limit how much waiting to be placed can be made. (stop production if exceed X)
- Make a even so if X or more is waiting first smoke, then if X2 is waiting to be placed it will make a zzt.. 
(thornsworth idea)

**Rimworld (please complete the following information):**
 - Type: Bug/function
 - Severity: Minor
 

**Additional context**

**Other ideas?**
I think a event like aprouch like thornsworth idea might be good for balance.
(maybe turn off in mod settings? (with a mod extention))

**Comment by zymex22** [comment]
merged into #168
